web/testing/attachment-api/index.html

```diff
@@ -2,79 +2,71 @@
-    <meta name="viewport" content="width=device-width,user-scalable=no" /> 
+
+    <!-- Set the viewport width to match phone and tablet device widths -->
+    <meta name="viewport" content="width=device-width,user-scalable=no" />
 
     <!-- Allow KeymanWeb to be saved to the iPhone home screen -->
     <meta name="apple-mobile-web-app-capable" content="yes" />
-    
+
     <!-- Enable IE9 Standards mode -->
-    <meta http-equiv="X-UA-Compatible" content="IE=edge" /> 
-      
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+
     <title>KeymanWeb Sample Page - Uncompiled Source</title>
 
-    <!-- Your page CSS --> 
-    <style type='text/css'>   
+    <!-- Your page CSS -->
+    <style type='text/css'>
       body {font-family: Tahoma,helvetica;}
       h3 {font-size: 1em;font-weight:normal;color: darkred; margin-bottom: 4px}
       .test {font-size: 1.5em; width:80%; min-height:30px; border: 1px solid gray;}
-      #KeymanWebControl {width:50%;min-width:600px;}       
-    </style> 
- 
-    <!-- Insert uncompiled KeymanWeb source scripts -->              
+      #KeymanWebControl {width:50%;min-width:600px;}
+    </style>
+
+    <!-- Insert uncompiled KeymanWeb source scripts -->
     <script src="../../release/unminified/web/keymanweb.js" type="application/javascript"></script>
 
-    <!-- 
+    <!--
       For desktop browsers, a script for the user interface must be inserted here.
-       
-      Standard UIs are toggle, button, float and toolbar.  
-      The toolbar UI is best for any page designed to support keyboards for 
+
+      Standard UIs are toggle, button, float and toolbar.
+      The toolbar UI is best for any page designed to support keyboards for
       a large number of languages.
     -->
     <script src="../../release/unminified/web/kmwuitoggle.js"></script>
- 
+
     <!-- Initialization: set paths to keyboards, resources and fonts as required -->
     <script>
-	// Thank you to https://stackoverflow.com/questions/22607150/getting-the-url-parameters-inside-the-html-page. 
-	var GetURLParameter = function(sParam) {
-		var sPageURL = window.location.search.substring(1);
-		var sURLVariables = sPageURL.split('&');
-		for (var i = 0; i < sURLVariables.length; i++) {
-			var sParameterName = sURLVariables[i].split('=');
-			if (sParameterName[0] == sParam) {
-				return sParameterName[1];
-			}
-		}
-	};
-	
-	var attachType = GetURLParameter("mode");
-	var attachText = attachType ? attachType : "default";
-	
-	var kmw=window.keyman;
-	kmw.init({
-		attachType: attachType ? attachType : '',
-	});
-    </script> 
-    
+      // Thank you to https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams
+      const attachType = (new URLSearchParams(window.location.search)).get("mode");
+      var attachText = attachType ? attachType : "default";
+
+      var kmw=window.keyman;
+      if (attachType) {
+        kmw.init({
+          attachType: attachType});
+      } else {
+        kmw.init({});
+      }
```

Comment:
Why change this from the previous, more compact version?

```
  kmw.init({
		attachType: attachType ? attachType : '',
	});
```

Reply:
At the time I did that, I was expecting attachType to be `auto` | `manual` | undefined.
To keep unit tests working, I put the empty string `''` back. I can revert this block too.